test/benchmarks/test_utility_scale.py

```diff
@@ -0,0 +1,71 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=no-member,invalid-name,missing-docstring,no-name-in-module
+# pylint: disable=attribute-defined-outside-init,unsubscriptable-object
+
+import os
+
+from qiskit import QuantumCircuit
+from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
+from qiskit.providers.fake_provider import GenericBackendV2
+from qiskit.transpiler import CouplingMap
+
+
+class UtilityScaleBenchmarks:
+    params = ["cx", "cz", "ecr"]
+    param_names = ["2q gate"]
+
+    def setup(self, basis_gate):
+        cmap = CouplingMap.from_heavy_hex(9)
+        basis_gates = ["rz", "x", "sx", basis_gate, "id"]
+        backend = GenericBackendV2(
+            cmap.size(), basis_gates, coupling_map=cmap, control_flow=True, seed=12345678942
+        )
+        self.pm = generate_preset_pass_manager(2, backend, seed_transpiler=1234567845)
+        qasm_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "qasm")
+        self.qft_qasm = os.path.join(qasm_dir, "qft_N100.qasm")
+        self.qft_qc = QuantumCircuit.from_qasm_file(self.qft_qasm)
+        self.square_heisenberg_qasm = os.path.join(qasm_dir, "square_heisenberg_N100.qasm")
+        self.square_heisenberg_qc = QuantumCircuit.from_qasm_file(self.square_heisenberg_qasm)
+        self.qaoa_qasm = os.path.join(qasm_dir, "qaoa_barabasi_albert_N100_3reps.qasm")
+        self.qaoa_qc = QuantumCircuit.from_qasm_file(self.qaoa_qasm)
+
+    def time_parse_qft_n100(self, _):
+        QuantumCircuit.from_qasm_file(self.qft_qasm)
+
+    def time_parse_square_heisenberg_n100(self, _):
+        QuantumCircuit.from_qasm_file(self.square_heisenberg_qasm)
+
+    def time_parse_qaoa_n100(self, _):
+        QuantumCircuit.from_qasm_file(self.qaoa_qasm)
```

Comment:
Can we change `QuantumCircuit.from_qasm_file` to `qasm2.load` for future-proofing the removal of the `QuantumCircuit` method?

Reply:
Sure, I'll update this.

FWIW, I typically still rely on this constructor method pretty heavily though so I'm not sure if we want to remove them fully (maybe rename it `from_qasm2_file`). Mostly because it's a shortcut to match what `qasm2.dumps()` returns and I don't have to remember the syntax for how to say the stdlib is qiskit's extended version and not the paper one.

Reply:
I still very very much want to update `qasm2.loads` and `dumps` to match each other (#10737), but I have no idea when I can prioritise that :(.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/12148/commits/80cd3c39ea2cacfbfa4488f6464a8b69e16d1d10